utils/modular_model_converter.py

```diff
@@ -1038,18 +1048,64 @@ def _recursively_add_all_new_needed_functions_in_files(self):
                 matching_callers = calling_entities & file_elements
                 added = self._maybe_add_function_to_body(top_level_function, body, function_node, matching_callers)
                 # If the function was added, we need to recursively add all its dependencies
+                builtin_functions = [
+                    "abs",
+                    "all",
+                    "any",
+                    "ascii",
+                    "bin",
+                    "bool",
+                    "bytearray",
+                    "bytes",
+                    "chr",
+                    "dict",
+                    "divmod",
+                    "enumerate",
+                    "filter",
+                    "float",
+                    "format",
+                    "frozenset",
+                    "hash",
+                    "hex",
+                    "int",
+                    "isinstance",
+                    "issubclass",
+                    "iter",
+                    "len",
+                    "list",
+                    "map",
+                    "max",
+                    "min",
+                    "next",
+                    "oct",
+                    "ord",
+                    "pow",
+                    "range",
+                    "repr",
+                    "reversed",
+                    "round",
+                    "set",
+                    "slice",
+                    "sorted",
+                    "str",
+                    "sum",
+                    "tuple",
+                    "type",
+                    "zip",
+                ]
                 if added:
                     for dependency, parent in find_all_dependencies(
                         top_level_function, self.function_call_dependency_mapping
                     ):
-                        self._maybe_add_function_to_body(
-                            dependency, body, self.all_definitions[dependency], parent=parent
-                        )
+                        if dependency not in builtin_functions and dependency in self.all_definitions:
```

Comment:
@ArthurZucker I added the second part of the check here to pass the test, it does not seem to create any missed imports so I let it but maybe it has unwanted consequences.

Reply:
Nice then thanks!

Reply:
Ha nice @aymeric-roucher, indeed I forgot about built-ins when adding this! Thanks for correcting it! If you only check for `if dependency in self.all_definitions`, it should be enough though no? Built-ins will never be added to `self.all_definitions` during visit, so it would avoid having the large list

Reply:
It's done! ✅